engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -1794,7 +1855,12 @@ public void setAvailableResources(int memoryMB)
   public void onStatusUpdate(PTOperator oper)
   {
     for (StatsListener l : oper.statsListeners) {
-      final StatsListener.Response rsp = l.processStats(oper.stats);
+      final StatsListener.Response rsp;
+      if (l instanceof StatsListener.StatsListenerWithContext) {
```

Comment:
Pushed one change to internally handled all listeners through new interface. operators implementing  the interface needs implement Serializable this might break backward compatibility as old operators implementing stats listeners will fail to serialize.  checking for another approach.